insights/core/dr.py

```diff
@@ -376,8 +361,119 @@ def register_component(delegate):
     MODULE_NAMES[component] = get_module_name(component)
     BASE_MODULE_NAMES[component] = get_base_module_name(component)
 
-    name = get_name(component)
-    COMPONENT_NAME_CACHE[name] = component
+
+class ComponentType(object):
+    """
+    ComponentType is the base class for all component type decorators.
+    """
+
+    requires = []
```

Comment:
Setting these names up as mutable state types here seems like a source of possible confusion and errors in the future.

Reply:
They're meant to be overridden in base classes, and the initialization of the base classes hide them. They're the "auto_requires" of the previous implementation.